x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/hooks/use_processing_simulator.ts

```diff
@@ -146,6 +147,14 @@ export const useProcessingSimulator = ({
     [liveDraftProcessors]
   );
 
+  const [memoizedSamplingCondition, setMemoizedSamplingCondition] = useState(samplingCondition);
+
+  useDeepCompareEffect(() => {
+    if (samplingCondition !== memoizedSamplingCondition) {
+      setMemoizedSamplingCondition(samplingCondition);
+    }
+  }, [samplingCondition]);
```

Comment:
I don't think we need a deep compare effect, you can run a similar comparison at render time and skip this extra side-effect:
```tsx
const [memoizedSamplingCondition, setMemoizedSamplingCondition] = useState(samplingCondition);
if (!isEqual(samplingCondition, memoizedSamplingCondition) {
  setMemoizedSamplingCondition(samplingCondition);
}
```

Reply:
maybe a stupid question, but why not calculate `samplingCondition` in the `useStreamsAppFetch` callback?

Reply:
Because the `samplingCondition` is derived from the `liveProcessors`, Joe added this memoization probably because the condition was recomputed on each live processor change, even if the condition didn't change at all.
If we calculate that inside the `useStreamsAppFetch` call, it would cause the same behaviour Joe wants to fix.
Good catch Joe!

Reply:
Well, we want to refetch samples if the condition actually changes, but in the old state it would update every time the `liveProcessors` change, which includes the user typing a pattern, which is not what we want.

Reply:
ah right, I see the change now, I thought the suggestion was to get rid of the deep equals comparison entirely.